### PR TITLE
use ID instead of login for flaky-test-retryer

### PR DESCRIPTION
**Pull request by TrevorFarrelly**
**What this PR does, why we need it**:
identify previous comments by user ID rather than login, so we can handle updates to the bot's display name and login info. Tested locally.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes to reviewers**:

**User-visible changes in this PR**:

### CODE REVIEW

File: tools/flaky-test-retryer/github_commenter.go [hunk 1]
```diff
@@ -59,7 +59,7 @@ func NewGithubClient(githubAccount string, dryrun bool) (*GithubClient, error) {
 	if err != nil {
 		return nil, err
 	}
-	return &GithubClient{ghc, *user.Login, dryrun}, nil
+	return &GithubClient{ghc, user.GetID(), dryrun}, nil
```
Comment (chaodaiG) [inline tools/flaky-test-retryer/github_commenter.go]: GetID return int64 instead of ptr, right?

**Comment by chaodaiG** [comment]
/ok-to-test
/lgtm
/approve

/hold
feel free unhold it if the answer to my question is yes

File: tools/flaky-test-retryer/github_commenter.go (hunk 1 above)
Comment (TrevorFarrelly) [reply tools/flaky-test-retryer/github_commenter.go]: Yes, seems the getter funcs strip the pointer off for us: https://godoc.org/github.com/google/go-github/github#User.GetID

**Pull request merged**